lib/icinga/checkable-comment.cpp

```diff
@@ -19,14 +19,15 @@ void Checkable::RemoveAllComments()
 	}
 }
 
-void Checkable::RemoveCommentsByType(int type, const String& removedBy)
+void Checkable::RemoveAckComments(const String& removedBy, double createdBefore)
 {
 	for (const Comment::Ptr& comment : GetComments()) {
-		/* Do not remove persistent comments from an acknowledgement */
-		if (comment->GetEntryType() == CommentAcknowledgement && comment->GetPersistent())
-			continue;
+		if (comment->GetEntryType() == CommentAcknowledgement) {
+			/* Do not remove persistent comments from an acknowledgement */
+			if (comment->GetPersistent() || comment->GetEntryTime() > createdBefore) {
+				continue;
+			}
```

Comment:
Looks strange from a style perspective. I'd either consistently nest everything inside one `if` or use `continue` for everything as both `if`s control whether the rest of the loop body is executed. The comment only applies to half of the condition at the moment, that could be improved with multiple `if`s and `continue`.